### PR TITLE
Larson-Miller based damage model

### PR DESCRIPTION
**Pull request by reverendbedford**
This PR adds a Larson-Miller damage model that directly converts a Larson Miller relation for time-to-rupture into damage using the time fraction approach.

### CODE REVIEW

File: src/damage_wrap.cxx [hunk 1]
```diff
@@ -11,8 +11,10 @@ PYBIND11_DECLARE_HOLDER_TYPE(T, std::shared_ptr<T>)
 namespace neml {
 
 PYBIND11_MODULE(damage, m) {
+  py::module::import("neml.models");
```
Comment (arovinelli) [inline src/damage_wrap.cxx]: is this required?
it doesn't seem to be used anywhere

File: examples/lm_damage_model.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  lmps = []
+
+  for s in srange:
+    res = drivers.creep(model, s, 100.0, 300000*3600.0, 
+        T = T, check_dmg = True, dtol = 0.95, nsteps_up = 20,
+        nsteps = 1000, logspace = True)
+    #plt.figure()
+    #plt.plot(res['rtime'], res['rstrain'])
+    #plt.show()
```
Comment (arovinelli) [inline examples/lm_damage_model.py]: if not used delete, otherwise add a comment about what is going to happen uncommented

File: examples/example.xml [hunk 3]
```diff
@@ -69,4 +69,67 @@
      </hardening>
    </flow>
  </powerlaw>
+
+  <gr91_simple type="SmallStrainCreepPlasticity">
```
Comment (arovinelli) [inline examples/example.xml]: not indented consistently with previous models

File: doc/sphinx/damage/larsonmiller.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+.. math::
+   \omega_{n+1} = \omega{n} + \frac{\Delta t_{n+1}}{t_R\left(\sigma_{eff} \left(1-\omega_{n+1}\right) \right)}
+
+where :math:`\sigma_{eff}` is a modular effective stress, defined by a :ref:`effective-stress` object, and :math:`t_R` is a time-to-rupture Larson-Miller relation provided
+by a :ref:`larson-miller` object.
+
+This model implements classical time-fraction damage summation in a finite 
+element framework.  It currently does not include graceful element deletion
+routines, though those may be added in the future.  This means the simulation
+can only progress up to the point of damage nucleation and not beyond.
```
Comment (arovinelli) [inline doc/sphinx/damage/larsonmiller.rst]: damage nucleation? isn't that the first time instance in which `omega>0`
I would rephrase saying that the simulation in NEML will not be able to reach `omega=1` because NEML doesn't have element deletion procedure.

Comment (arovinelli) [reply doc/sphinx/damage/larsonmiller.rst]: Also, please add a reference to the Modular creep damage page for the use of ekill dkill and sfact

File: src/damage_wrap.cxx (hunk 1 above)
Comment (reverendbedford) [reply src/damage_wrap.cxx]: It is definitely required on the python side.

File: examples/lm_damage_model.py (hunk 2 above)
Comment (reverendbedford) [reply examples/lm_damage_model.py]: Yep, that's a good catch.

File: examples/example.xml (hunk 3 above)
Comment (reverendbedford) [reply examples/example.xml]: Yep.

File: doc/sphinx/damage/larsonmiller.rst (hunk 4 above)
Comment (reverendbedford) [reply doc/sphinx/damage/larsonmiller.rst]: Sure.
